### DISCUSSION TITLE
improve template editor: add Clear button/always show 2 lines

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#8739] [Other feature requests]
Template editor can be a difficult instrument especially on Mobile devices. 
Selecting all text to replace it with a new template, or clear to start all over is very hard, no to say impossible. A Clear button could facilitate this process tremendously.

Secondly, selecting text in a single line in the editor is equally difficult. Desktop Mouse is already hard (the scroll bar covering the single line in the editor almost completely) on Mobile this is even more difficult. Adding a minimum line count of 2 could facilitate this, allowing some maneuvering space between the single line and the controls of the editor.

<img width="528" alt="Schermafbeelding 2021-03-28 om 15 46 01" src="https://user-images.githubusercontent.com/33354141/112754522-bb20d400-8fdc-11eb-8e86-9d737fcf083e.png">

Thanks for considering

**Comment by Mariusthvdb** [comment]
might I post a slight nudge here? issue is still very much alive while on 2021.10

**Comment by spacegaier** [comment]
The scrollbar should not be overlaying the editor. Please create a bug ticket for that. Might be something that has to be fixed upstream actually. Seem platform specific, because it works fine for me on Windows.

**Reply by GrizzlyAK** [reply]
A GIANT +1 for adding a Clear button. Or better yet, just bring it up BLANK and add an "Example" button. Who actually ever uses any text that is in the example anyway? It was helpful the first two times using it, but now the example is an annoyance. Bring it up ready to use. That's what it is for.

**Comment by Mariusthvdb** [comment]
ok will do, and just checked, still happening on Desktop Mac, HA 2021.11.0.dev20211026:

<img width="848" alt="Schermafbeelding 2021-10-26 om 17 33 46" src="https://user-images.githubusercontent.com/33354141/138912293-af384b52-b27d-4b06-b067-91a429bafb09.png">

**Reply by Mariusthvdb** [reply]
this is still happening . as a matter of fact, I had forgotten about this issue, and made a new one....

**Comment by Daniel-dev22** [comment]
I didn't realize there was a place for this on GitHub.

Here's an accompanying feature request I created from the forums. 

https://community.home-assistant.io/t/move-reset-to-demo-template-somewhere-else/432132?u=danny2100

**Comment by Mariusthvdb** [comment]
> The scrollbar should not be overlaying the editor. Please create a bug ticket for that. Might be something that has to be fixed upstream actually.

https://github.com/home-assistant/frontend/issues/16929

**Comment by karwosts** [comment]
On my android mobile if I just press and hold on the template editor, I get a little popup with the option "select all". If I can select all, then a single press of the delete key clears the editor. 

Not to dismiss the option for a clear button, but just pointing out that's one way to clear it with the current functionalty. I'm not sure if there's an iOS equivalent.

**Reply by Daniel-dev22** [reply]
Another issue here with the template editor is it's way too easy on mobile to hit reset to demo template.

**Reply by Mariusthvdb** [reply]
> On my android mobile if I just press and hold on the template editor, I get a little popup with the option "select all". If I can select all, then a single press of the delete key clears the editor. 
> I'm not sure if there's an iOS equivalent. 


Ofc we tried that, but below is what we get, and it's even more of a horrific UX to try and pull that select box over the full Text window....
![image](https://github.com/home-assistant/frontend/assets/33354141/2dec7d3d-196f-490e-8e7e-1ced8ea4414a)

**Reply by spacegaier** [reply]
I created a PR that would add the button plus a confirmation dialog to prevent accidental data loss: https://github.com/home-assistant/frontend/pull/17020

**Comment by Mariusthvdb** [comment]
the clear button is functioning nicely!

please dont forget the bottom line issue, which is still very much alive in the HA frontend(also now in the new template helper 
i made a separate issue for that here https://github.com/home-assistant/frontend/issues/16929
